package.json

```diff
@@ -6,8 +6,8 @@
   "dependencies": {
     "@material-ui/core": "^4.9.1",
     "@material-ui/icons": "^4.9.1",
-    "@twilio-labs/plugin-rtc": "^0.8.2",
-    "@twilio/conversations": "^1.1.0",
+    "@twilio-labs/plugin-rtc": "0.8.2",
```

Comment:
The carats were removed since there were some breaking changes in the newer versions of `@twilio-labs/plugin-rtc` and `@twilio/conversations` that prevented the app from running

Reply:
Interesting. What errors were you seeing?

The version numbers of `@twilio-labs/plugin-rtc` and `@twilio/conversations` as specified in the package-lock.json file are 0.8.2 and 1.1.0, respectively. package-lock.json dictates the exact versions that should be installed. So I'm wondering if you ran into some other kind of issue, because 
npm shouldn't have installed newer versions. 

I consider it a best practice to leave the carets in (with exception of the react-scripts package). The version numbers listed in package.json describe the range of versions of a package that will be compatible with the app. Note that the package.json file does not dictate which specific version is installed (this is what package-lock.json is for). The caret means that this app can use any package version that has a minor or patch version number (but not major version number) higher than what's specified. The three numbers in a version all have a specific meaning, which is called [semantic versioning](https://docs.npmjs.com/about-semantic-versioning). Generally speaking, any new package version that only increments the minor or patch version numbers should not introduce any breaking changes. 

The `package-lock.json` file serves a different purpose. This file dictates the exact package versions that should be installed. This is to ensure that each time anybody runs `npm install`, the exact same set of packages is installed. This helps reduce weird issues where some users experience bugs that other users don't because they installed a slightly different set of packages. 

So the caret won't introduce any breaking changes, and it also is meant to indicate that newer minor and patch versions can be installed. So let's keep the carets in (and also add it to rimraf, copyfiles, twilio, and @twilio/video-processors). Let me know if there are any issues that you run into. I'd be more than happy to take a look. 🙂 

Also, this [semver calculator](https://semver.npmjs.com/) is a fun way to learn about npm's version syntax.

Reply:
It's been a while since Charlie and I dealt with these issues, so I don't remember them off the top of my head. Thinking back, I think I got those errors after updating all my npm packages, which introduced the breaking changes from the newer versions. I was able to add back the carats with no issues though!